src/convertToObject.js

```diff
@@ -1,12 +1,30 @@
 'use strict';
 
+const toCamelCase = require('./toCamelCase');
+
 /**
  * @param {string} sourceString
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const splitString = sourceString.split(';');
+  const newObject = {};
+
+  splitString.forEach((rule) => {
+    if (rule.trim()) {
+      const [key, value] = rule.split(':');
+
+      const formattedKey = key.includes('-')
+        ? key.trim()
+        : toCamelCase(key.trim());
+
+      newObject[formattedKey] = value.trim();
+    }
+  });
+
+  return newObject;
```

Comment:
The function `convertToObject` directly modifies the `newObject` within the `forEach` loop. According to the checklist, you should avoid directly modifying an existing object within an iteration method. Consider using a functional approach with `reduce` to accumulate the properties into a new object.